app/lib/task/backend.dart

```diff
@@ -341,6 +328,34 @@ class TaskBackend {
     }
   }
 
+  Future<void> _runOneScanPackagesUpdate({
+    required bool Function() isAbortedFn,
+  }) async {
+    final next = await calculateScanPackagesUpdatedLoop(
+      _scanPackagesUpdatedState,
+      _db.packages.listUpdatedSince(_scanPackagesUpdatedState.since),
+      isAbortedFn,
+    );
+    _scanPackagesUpdatedState = next.state;
+
+    for (final p in next.packages) {
+      if (isAbortedFn()) {
+        return;
+      }
+      // Check the package
+      await trackPackage(p, updateDependents: true);
+    }
+  }
+
+  @visibleForTesting
+  Future<void> runOneIterationOfBackgroundLoops() async {
```

Comment:
Is this ever called?

Reply:
Why not just expose the actual _runOneScanPackagesUpdate function to tests? It seems more honest to test that than some wrapper that is otherwise not used in production.